### PR TITLE
Fixes #320 Email buildstatus on process completion

### LINKED ISSUE #320: Add option to email the build report on completion of docs generation (similar to travis)
### This is a
- [ ] Bug report
- [x] Feature request

### Current behaviour

A user cannot know the status of docs generation (Whether it was successful or not.)
### Expected behaviour

Implement the feature similar to Travis.
image

### PR DESCRIPTION
**Pull request by imujjwal96**
## Description

Mail is sent:
* on the first build
* on every failed build
* on the change of build status (true->false or false->true)
* to the person who registered the repository
## Related Issue

issue #320

## How Has This Been Tested?

https://yaydoc7.herokuapp.com
## Screenshots (if appropriate):
image

## Types of changes

- [ ] Bug fix 
- [x] New feature

## Checklist:

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only one commit per issue.

### CODE REVIEW

**Comment by imujjwal96** [comment]
@pri22296  @sch00lb0y  Please review

File: backend/mailer.js [hunk 1]
```diff
@@ -53,5 +54,35 @@ exports.sendEmail = function (data) {
       console.log('Message sent: ' + info.response);
     }
   });
+};
+
+exports.sendMailOnBuild = function (buildStatus, email, repository) {
+  var options = {
```
Comment (sch00lb0y) [inline backend/mailer.js]: use this as a global `const` variable

Comment (imujjwal96) [inline backend/mailer.js]: Updated

**Comment by imujjwal96** [comment]
@pri22296 @sch00lb0y  Please review.

File: model/repository.js [hunk 2]
```diff
@@ -10,6 +10,8 @@ const repositorySchema = new mongoose.Schema({
     login: String
   },
   accessToken: String,
+  buildStatus: Boolean,
+  mailService: Boolean,
```
Comment (sch00lb0y) [inline model/repository.js]: any reason for `mailService`  ?

Comment (imujjwal96) [inline model/repository.js]: User should have an option to disable build status mail. It can be considered spamming.

**Comment by imujjwal96** [comment]
@pri22296  @sch00lb0y  Please review.

**Comment by pri22296** [comment]
I registered earlier with the CI. Then I removed the webhook from my registered repo. Now registering again cause this.

screenshot 196

**Comment by imujjwal96** [comment]
@pri22296 please raise a separate issue :)

**Pull request merged**